.travis.yml

```diff
@@ -52,6 +53,13 @@ before_script:
       make install &&
       popd;
     fi
+  - if [[ ${AMPI} ]]; then
+      git clone --depth=1 https://github.com/UIUC-PPL/charm && pushd charm &&
+      ./build AMPI netlrts-linux-x86_64 ${CC} -j2 && export CXX=$PWD/bin/ampicxx && 
+      CMAKE_OPTS+=( -DMPIEXEC="$PWD/bin/ampirun" -DMPIEXEC_POSTFLAGS="+vp4" ) && 
```

Comment:
do we need to change the way the number of ranks are set using the `+p2` syntax as suggested?

Reply:
@sslattery with AMPI you call a job like this: `ampirun -np2 ./pgm +vp4`, so it runs on 2 cores, but uses 4 virtual processors (the `+vp` option).

Reply:
Hence we put `+vp` in the `MPIEXEC_POSTFLAGS` instead of the `MPIEXEC_PREFLAGS`.